### DISCUSSION TITLE
Remember window location

### DISCUSSION THREAD

**Discussion by joepio** [tauri-apps/tauri#2954] [Q&A] [answered]
When I open my browser, it remembers which tabs were open, and which one was selected. I'd love to have a similar experience with Tauri.

Is there a way to store and retrieve the last location and open a window there?

## Use JS + `window.location` + localstorage

When the app initializes, check some predefined key set in localstorage. If there's a key, go there.
When the page changes, update this value.

This method seems a bit hacky. Do you have other suggestions?

**Comment by FabianLars** [comment, marked as answer]
Nah, that's pretty much the way to go. I'm not sure if you'd want to use localstorage for that tho, sounds too easy to manipulate.
If you're not using a router == If you're using "normal" navigation methods, you can also use the `.on_page_load` hook in rust, so you potentially wouldn't need to use js at all (other than navigating in already spawned windows).

**Comment by probablykasper** [comment]
In case anyone stumbles upon this, [tauri-plugin-window-state](https://github.com/tauri-apps/tauri-plugin-window-state) can be used for this

**Comment by gschier** [comment]
> In case anyone stumbles upon this, [tauri-plugin-window-state](https://github.com/tauri-apps/tauri-plugin-window-state) can be used for this

This seems to only remember the following properties, not `window.location` (current URL):

```
        const SIZE        = 1 << 0;
        const POSITION    = 1 << 1;
        const MAXIMIZED   = 1 << 2;
        const VISIBLE     = 1 << 3;
        const DECORATIONS = 1 << 4;
        const FULLSCREEN  = 1 << 5;
```
